aikau/src/main/resources/alfresco/lists/SelectedItemStateMixin.js

```diff
@@ -0,0 +1,261 @@
+ * selection is typically actioned by either individual [Selector]{@link module:alfresco/renderers/Selector}
+ * widgets in a [list]{@link module:alfresco/lists/AlfList} or by a menu action such as the ones provided by
+ * the [AlfSelectDocumentListItems]{@link module:alfresco/documentlibrary/AlfSelectDocumentListItems}.</P>
+ * <p>Mixing modules should call the 
+ * [createSelectedItemSubscriptions]{@link module:alfresco/lists/SelectedItemStateMixin#createSelectedItemSubscriptions}
+ * on creation to keep track of selected items. The can also call the 
+ * [publishSelectedItems]{@link module:alfresco/lists/SelectedItemStateMixin#publishSelectedItems} function
+ * to broadcast the details of changes to item selection.</p>
+ * 
+ * @module alfresco/lists/SelectedItemStateMixin
+ * @author Dave Draper
+ * @since 1.0.39
+ */
+define(["dojo/_base/declare",
+        "alfresco/core/Core",
+        "alfresco/core/topics",
+        "alfresco/documentlibrary/_AlfDocumentListTopicMixin",
+        "dojo/_base/lang"], 
+        function( declare, Core, topics, _AlfDocumentListTopicMixin, lang) {
+   
+   return declare([Core, _AlfDocumentListTopicMixin], {
+      
+      /**
+       * This is used to keep track of the documents that are currently selected. It is initialised to an empty
+       * array in the constructor, [onItemSelected]{@link module:alfresco/lists/SelectedItemStateMixin#onItemSelected} 
+       * adds elements and the [onItemDeselected]{@link module:alfresco/lists/SelectedItemStateMixin#onItemDeselected} 
+       * removes them.
+       *
+       * @instance
+       * @type {object}
+       * @default
+       */
+      currentlySelectedItems: null,
+
+      /**
+       * The number of milliseconds to wait after a selection event before publishing the latest selected item
+       * data.
+       *
+       * @instance
+       * @type {number}
+       * @default
+       */
+      debounceTime: 50,
+
+      /**
+       * Indicates whether or not the the [disable]{@link module:alfresco/lists/SelectedItemStateMixin#disable}
+       * function should be called when no items are selected.
+       *
+       * @instance
+       * @type {boolean}
+       * @default
+       */
+      disableWhenNothingSelected: false,
+
+      /**
+       * This is the dot-notation addressed property within the selection/de-selection publication payload that
+       * uniquely identifies the item.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       */
+      itemKeyProperty: "node.nodeRef",
+
+      /**
+       * An array of the [itemKeyProperty]{@link module:alfresco/lists/SelectedItemStateMixin#itemKeyProperty}
+       * attributes taken from each entry in the 
+       * [currentlySelectedItems]{@link module:alfresco/lists/SelectedItemStateMixin#currentlySelectedItems}
+       * map.
+       * 
+       * @instance
+       * @type {string[]}
+       * @default
+       */
+      selectedItems: null,
```

Comment:
currentlySelectedItems is an *object*.... selectedItems is an *array of objects*